apps/member/src/hooks/queries/support/useSupportListUi.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router';
+
+import { Support } from '@type/support';
+
+interface UseSupportListUIProps {
+  sortedData: Support[];
+  supportData: Support[];
+  page: number;
+  size: number;
+  handlePageChange: (p: number) => void;
+  showAll: boolean;
+}
+
+export const useSupportListUI = ({
+  sortedData,
+  supportData,
+  page,
+  size,
+  handlePageChange,
+  showAll,
+}: UseSupportListUIProps) => {
+  const [searchParams] = useSearchParams();
+  const selectedId = searchParams.get('selected');
+
+  const [currentOpenItemIndex, setCurrentOpenItemIndex] = useState(-1);
+
+  const handleTableItemClick = useCallback((id: number) => {
+    setCurrentOpenItemIndex(id);
+  }, []);
+
+  useEffect(() => {
+    setCurrentOpenItemIndex(-1);
+
+    if (selectedId && showAll) {
+      const targetId = parseInt(selectedId);
+      const targetSupport = supportData.find((item) => item.id === targetId);
+
+      if (targetSupport) {
+        const targetIndex = sortedData.findIndex(
+          (item) => item.id === targetId,
+        );
+        const targetPage = Math.floor(targetIndex / size) + 1;
+
+        if (targetPage !== page) {
+          handlePageChange(targetPage);
+        } else {
+          setCurrentOpenItemIndex(targetId);
+        }
+      }
+    }
```

Comment:
`targetSupport`의 경우 `find` 연산을 통해 아이템 존재 여부만 확인하고 해당 결과로 반환되는 객체는 사용하지 않고 있는 것 같아요. 이런 경우 `find` 연산보다 [`some`](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/some) 연산을 사용하는 것이 더 적절해 보여요.

그리고 지금 코드 같은 경우 early return 패턴을 사용하면 코드 흐름이 더욱 단순해질 것 같아 제안 드립니다! [early return 패턴의 장단점](https://thearchivelog.dev/article/are-early-returns-any-good/)
```suggestion
  setCurrentOpenItemIndex(-1);
  
  if (!selectedId || !showAll) return;
  const targetId = Number(selectedId); 
  
  const existsInSupportData = supportData.some(item => item.id === targetId);
  if (!existsInSupportData) return;

  const targetIndex = sortedData.findIndex(item => item.id === targetId);
  if (targetIndex === -1) return;

  const targetPage = Math.floor(targetIndex / size) + 1;

  if (targetPage !== page) {
    handlePageChange(targetPage);
  } else {
    setCurrentOpenItemIndex(targetId);
  }
```